src/Value.php

```diff
@@ -138,4 +141,17 @@ public function asNonEmptyStrings(): array
 
         return array_values($this->value);
     }
+
+    public function asDateTimeImmutable(): DateTimeImmutable
```

Comment:
IMO it makes sense to add an optional format here and use `createFromFormat`, what do you think?

Also please add new method to readme.

Reply:
Added :+1: